packages/appimage/AppImageBuilder.yml

```diff
@@ -0,0 +1,106 @@
+# appimage-builder recipe see https://appimage-builder.readthedocs.io for details
+version: 1
+
+script:
+  # Download, configure and make
+#  - git clone https://github.com/ygrek/mldonkey.git
+#  - cd mldonkey
+#  - ./configure --enable-batch --enable-upnp-natpmp --disable-directconnect --disable-fasttrack --disable-gnutella --disable-gnutella2 --enable-gui=newgui2 -->
+#  - make
+  # Remove any previous build
+  - rm -rf AppDir | true
+  # Get application version
+#  - ./mlnet+gui -v 2>&1 | sed -ne "s#^.*Starting MLDonkey \([^ ]\+\) .*#APP_VERSION=\1#p" >> $BUILDER_ENV
+#  - cat "$BUILDER_ENV"
+#  - APP_VERSION=$("$SOURCE_DIR"/mlnet+gui -v 2>&1 | sed -ne "s#^.*Starting MLDonkey \([^ ]\+\) .*#\1#p")
+#  - echo "$APP_VERSION"
+  # Make some needed dirs
+  - mkdir -p "$TARGET_APPDIR"/usr/bin
+  - mkdir -p "$TARGET_APPDIR"/usr/share/icons/default
+  - mkdir -p "$TARGET_APPDIR"/usr/share/applications
+  # Copy the application data into the AppDir
+  - cp "$SOURCE_DIR"/mlnet+gui                         "$TARGET_APPDIR"/usr/bin/
+  - cp "$SOURCE_DIR"/distrib/mldonkey.desktop          "$TARGET_APPDIR"/usr/share/applications
+  - cp "$SOURCE_DIR"/icons/rsvg/type_source_normal.svg "$TARGET_APPDIR"/usr/share/icons/default/mldonkey.svg
+  # Relocatable by bruteforce patching
+  - sed -i -e 's#/usr#././#g' "$TARGET_APPDIR"/usr/bin/mlnet+gui
+  # Create symlink
+  - ln -s ./usr/bin/mlnet+gui "$TARGET_APPDIR"/mldonkey_gui
+
+AppDir:
+  path: ./AppDir
+  app_info:
+    id: mldonkey
+    name: mldonkey
+    icon: mldonkey
+#    version: {{APP_VERSION}}
+#    version: ${APP_VERSION}
+#    version: $APP_VERSION
+    version: 3.1.7.git
```

Comment:
this should be determined dynamically or filled in by configure

Reply:
The preceding commented lines were my attempts to to find how to extract the current version from the environment variable extracted and assigned in a previous process just above.

https://github.com/lulol/mldonkey/blob/81d5f0f82ea9b4a9d4c15cc72548bfbd06927da2/packages/appimage/AppImageBuilder.yml#L12-L16

But or there was some step missing in the documentation or I didn't understand how must be implemented so had to temporarily hardcode the current version.

Reply:
Implemented version setting with exported environment variables since the recipe methods doesn't seem to work.